packages/core/content-type-builder/admin/src/utils/parseDateValue.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema } from '@strapi/types';
+
+export const parseDateValue = <TAttribute extends Schema.Attribute.AnyAttribute>(
+  value: Schema.Attribute.Value<TAttribute>
+) => {
+  if (value instanceof Date && isValidDate(value)) {
+    return value;
+  }
+
+  if (typeof value === 'string' || typeof value === 'number') {
+    const date = new Date(value);
+    if (isValidDate(date)) {
+      return date;
+    }
+  }
```

Comment:
Couldn't this be simplified to the following?
```suggestion
  const date = new Date(value);
  
  return isValidDate(date) ? date : undefined;
```


Since the Date constructor don't throw with invalid data types (and instead returns an invalid date instance)

Reply:
oops, didn't see Bassel comments, could also be

```typescript
  const date = new Date(value);
  
  if (isValidDate(date)) {
    return date;
  }
```